python/raydp/tf/estimator.py

```diff
@@ -75,168 +77,200 @@ def __init__(self,
+                  "metrics": self._metrics,
+                  "callbacks": self._callbacks}
 
-        if evaluate_ds is not None:
-            evaluate_ds = evaluate_ds.batch(self._batch_size)
-            evaluate_tf_ds = self._create_tf_ds(evaluate_ds)
+        train_ds_pipeline = train_ds.repeat()
+        if self._shuffle:
+            train_ds_pipeline = train_ds_pipeline.random_shuffle_each_window()
+        dataset = {"train": train_ds_pipeline}
+        if evaluate_ds is None:
+            config["evaluate"] = False
         else:
-            evaluate_tf_ds = None
-
-        def data_creator(config):
-            if "TF_CONFIG" in os.environ:
-                tf_config = json.loads(os.environ["TF_CONFIG"])
-                world_rank = tf_config["task"]["index"]
-            else:
-                world_rank = -1
-            batch_size = config["batch_size"]
-            get_shard_config = config.get("get_shard", {})
-            if "shuffle" in config:
-                get_shard_config["shuffle"] = config["shuffle"]
-            train_data = train_tf_ds.get_shard(
-                world_rank, **get_shard_config).repeat().batch(batch_size)
-            options = tf.data.Options()
-            options.experimental_distribute.auto_shard_policy = \
-                tf.data.experimental.AutoShardPolicy.OFF
-            train_data = train_data.with_options(options)
-            evaluate_data = None
-            if evaluate_tf_ds is not None:
-                evaluate_data = evaluate_tf_ds.get_shard(
-                    world_rank, **get_shard_config).batch(batch_size)
-                evaluate_data = evaluate_data.with_options(options)
-            return train_data, evaluate_data
-
-        self._trainer = TFTrainer(model_creator=model_creator,
-                                  data_creator=data_creator,
-                                  num_replicas=self._num_workers,
-                                  **self._extra_config)
-        for i in range(self._num_epochs):
-            stats = self._trainer.train()
-            print(f"Epoch-{i}: {stats}")
-
-        if evaluate_tf_ds is not None:
-            print(self._trainer.validate())
+            evaluate_ds_pipeline = evaluate_ds.repeat()
+            if self._shuffle:
+                evaluate_ds_pipeline = evaluate_ds_pipeline.random_shuffle_each_window()
+            dataset["evaluate"] = evaluate_ds_pipeline
+        self._trainer.start()
+        results = self._trainer.run(
+            train_func=TFEstimator.train_fun,
+            dataset=dataset,
+            config=config,
+        )
+        return results
 
     def fit_on_spark(self,
                      train_df: DF,
                      evaluate_df: OPTIONAL_DF = None,
-                     fs_directory: Optional[str] = None,
-                     compression: Optional[str] = None) -> NoReturn:
+                     max_retries=3,
+                     stop_spark_after_conversion=False) -> NoReturn:
```

Comment:
Can we still support `fs_directory` and `compression` parameters so users can persist the data to disk in some cases? Let's keep the parameters but we can implement it in a followup PR.